include/clad/Differentiator/DynamicGraph.h

```diff
@@ -110,21 +110,46 @@ template <typename T> class DynamicGraph {
 
   /// Print the nodes and edges in the graph.
   void print() {
-    // First print the nodes with their insertion order.
-    for (const T& node : m_nodes) {
-      std::pair<bool, int> nodeInfo = m_nodeMap[node];
-      std::cout << (std::string)node << ": #" << nodeInfo.second;
-      if (m_sources.find(nodeInfo.second) != m_sources.end())
-        std::cout << " (source)";
-      if (nodeInfo.first)
-        std::cout << ", (done)\n";
-      else
-        std::cout << ", (unprocessed)\n";
+    std::unordered_set<size_t> visited;
+
+    // Recursive function to print nodes with indentation based on depth.
+    std::function<void(size_t, int)> printNode = [&](size_t nodeId, int depth) {
```

Comment:
```suggestion
    auto printNode = [&](size_t nodeId, int depth) {
```

I believe we can the recursion with the cheaper iteration approach.

Reply:
Yes

Reply:
@Abhinavcode13, ping.